src/main/java/jenkins/plugins/publish_over_ssh/BapSshHostConfiguration.java

```diff
@@ -42,51 +46,71 @@
 import org.apache.commons.logging.LogFactory;
 import org.kohsuke.stapler.DataBoundConstructor;
 
-import com.jcraft.jsch.ChannelSftp;
-import com.jcraft.jsch.JSch;
-import com.jcraft.jsch.JSchException;
-import com.jcraft.jsch.Session;
-import com.jcraft.jsch.SftpException;
-
 @SuppressWarnings("PMD.TooManyMethods")
 public class BapSshHostConfiguration extends BPHostConfiguration<BapSshClient, BapSshCommonConfiguration>
-                                                                                        implements Describable<BapSshHostConfiguration> {
+        implements Describable<BapSshHostConfiguration> {
 
     private static final long serialVersionUID = 1L;
     public static final int DEFAULT_PORT = 22;
     public static final int DEFAULT_TIMEOUT = 300000;
     public static final String CONFIG_KEY_PREFERRED_AUTHENTICATIONS = "PreferredAuthentications";
     private static final Log LOG = LogFactory.getLog(BapSshHostConfiguration.class);
+    public static final String HTTP_PROXY_TYPE = "http";
+    public static final String SOCKS_4_PROXY_TYPE = "socks4";
+    public static final String SOCKS_5_PROXY_TYPE = "socks5";
 
     private int timeout;
     private boolean overrideKey;
     private boolean disableExec;
     private final BapSshKeyInfo keyInfo;
 
+    private String proxyType;
+    private String proxyHost;
+    private int proxyPort;
+    private String proxyUser;
+    private String proxyPassword;
+
     // CSOFF: ParameterNumberCheck
     @SuppressWarnings("PMD.ExcessiveParameterList") // DBC for you!
     @DataBoundConstructor
     public BapSshHostConfiguration(final String name, final String hostname, final String username, final String encryptedPassword,
```

Comment:
DataBoundSetters have been introduced in 1.535 (https://groups.google.com/d/msg/jenkinsci-dev/58-DEvuJZWI/5QrxBZRFJ6IJ) but this project depends on 1.532

maybe the upgrade should be done on a separate PR to keep it clear